[docs] added "--enable_automatic_tablet_splitting" flag (#17419)
yugabyte/yugabyte-db: docs/content/preview/reference/configuration/yb-master.md

The number of shards per YB-TServer for each YCQL table when a user table is created.

> Default: `-1` (the value is calculated at runtime).

Reviewer: I'm not clear on whether this is all discussing defaults - if so, clarify as follows

The default is set depending on the type of deployment, as follows:
- For standard deployments, `-1`, which indicates that the the value is calculated at runtime as follows:
  - For servers with up to two CPU cores, the value is set to `4`.
  - For three or more CPU cores, the value is set to `8`.
- For local clusters created with `yb-ctl` or `yb-docker-ctl`, `2`.
- For clusters created with `yugabyted`, `1`.

Reviewer 2: That's not correct, the Default is `-1` and depending on the deployment (actually num of CPUs and tablet splitting value) a server is considering on-the-fly which value should be use (before 2.18 this considered value is set as a flag's value replacing `-1` -- which is not correct enough and that's why this was changed).
These:
For local clusters created with yb-ctl or yb-docker-ctl, 2
For clusters created with yugabyted, 1
have no relation to Default.

Reviewer: OK - the way it was phrased I couldn't tell. So what are they - advice?

Reviewer 2: Current version is looking OK from my perspective.

\
For servers with up to two CPU cores, the default value is considered as `4`.
\
For three or more CPU cores, the default value is considered as `8`.
\
Local cluster installations created with `yb-ctl` and `yb-docker-ctl` use a value of `2` for this flag.
\

> Clusters created with `yugabyted` use a default value of `1`.

Reviewer 2: Makes sense to add an empty line before this section as it is not related to explanation of _Default_

{{< note title="Note" >}}

- This value must match on all `yb-master` and `yb-tserver` configurations of a YugabyteDB cluster.

> - If the shard value is set to *Default* (`-1`), then the server internally *updates* the flag with intended value during startup prior to version `2.18` and the flag remains *unchanged* starting from version `2.18`.

Reviewer 2: These are actually "Important" notes, and not just notes. Also having 4 different note blocks is looking fine to me 
<img width="867" alt="image" src="https://github.com/yugabyte/yugabyte-db/assets/79331145/7844993d-2740-4aed-9ac3-ce1f6d670978">

cc @rthallamko3 to comment

Reviewer 2: Suggested update:
<img width="864" alt="image" src="https://github.com/yugabyte/yugabyte-db/assets/79331145/065fb465-01fc-4509-a9f3-bf45b3056a0a">

> - If the shard value is set to *Default* (`-1`), then the server internally *updates* the flag with intended value during startup prior to version `2.18` and the flag remains *unchanged* starting from version `2.18`.

Reviewer: ```suggestion
- If the flag is set to `-1`, on versions prior to 2.18, the server internally *updates* the flag with the intended value during startup; on version 2.18 and later, the flag remains *unchanged*.
```

> - The [`CREATE TABLE ... WITH TABLETS = <num>`](../../../api/ycql/ddl_create_table/#create-a-table-specifying-the-number-of-tablets) clause can be used on a per-table basis to override the `yb_num_shards_per_tserver` value.

Reviewer: ```suggestion
- Use the [`CREATE TABLE ... WITH TABLETS = <num>`](../../../api/ycql/ddl_create_table/#create-a-table-specifying-the-number-of-tablets) clause to override the `yb_num_shards_per_tserver` value when creating tables.
```

> - If [enable_automatic_tablet_splitting](#enable-automatic-tablet-splitting) is `true`, the default value is considered as `1` and tables will begin with 1 tablet *per node*; from version `2.18` for servers with up to 4 CPU cores, the value *is not defined* and tables will begin with 1 tablet (for servers with up to 2 CPU cores) or 2 tablets (for servers with up to 4 CPU cores) *per cluster*.

Reviewer: ```suggestion
- If [enable_automatic_tablet_splitting](#enable-automatic-tablet-splitting) is `true`, the default value is considered as `1` and tables are created with 1 tablet *per node*; for version 2.18 and later, for servers with up to 4 CPU cores, the value *is not defined* and tables are created with 1 tablet (for servers with up to 2 CPU cores) or 2 tablets (for servers with up to 4 CPU cores) *per cluster*.
```

Default: `true`

Note that this value must match on all `yb-master` and `yb-tserver` configurations of a YugabyteDB cluster.

> 

Reviewer 2: Maybe worth wrapping into Important block. Is there any new style to not use note block as for 2.14?

Author: Not really , I can include it back into a note box for getting it more attention. Thanks.
